src/ParallelAlgorithms/LinearSolver/Schwarz/Actions/ResetSubdomainSolver.hpp

```diff
@@ -0,0 +1,95 @@
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+struct GlobalCache;
+}  // namespace Parallel
+namespace tuples {
+template <typename...>
+struct TaggedTuple;
+}  // namespace tuples
+/// \endcond
+
+namespace LinearSolver::Schwarz::Actions {
+
+/*!
+ * \brief Reset the subdomain solver, clearing its caches related to the linear
+ * operator it has solved so far.
+ *
+ * Invoke this action when the linear operator changed. For example, an operator
+ * representing the linearization of a nonlinear operator changes whenever the
+ * point around which it is being linearized gets updated, i.e. in every
+ * iteration of a nonlinear solve. Note that the operator does _not_ change
+ * between iterations of a linear solve, so make sure you place this action
+ * _outside_ the looping action list for the linear solve.
+ *
+ * \par Skipping the reset:
+ * Depending on the subdomain solver in use, the reset may incur significant
+ * re-initialization cost the next time the subdomain solver is invoked. See the
+ * `LinearSolver::Serial::ExplicitInverse` solver for an example of a subdomain
+ * solver with high initialization cost. For this reason the reset can be
+ * skipped with the option
+ * `LinearSolver::Schwarz::Tags::SkipSubdomainSolverResets`. Skipping resets
+ * means that caches built up for the linear operator are never cleared, so
+ * expensive re-initializations are avoided but the subdomain solves may be
+ * increasingly inaccurate or slow down as the linear operator changes over
+ * nonlinear solver iterations. Whether or not skipping resets helps with the
+ * overall convergence of the solve is highly problem-dependent. A possible
+ * optimization would be to decide at runtime whether or not to reset the
+ * subdomain solver.
+ */
+template <typename OptionsGroup>
+struct ResetSubdomainSolver {
+  using const_global_cache_tags = tmpl::list<
+      LinearSolver::Schwarz::Tags::SkipSubdomainSolverResets<OptionsGroup>,
+      logging::Tags::Verbosity<OptionsGroup>>;
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            size_t Dim, typename ActionList, typename ParallelComponent>
+  static std::tuple<db::DataBox<DbTagsList>&&> apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ElementId<Dim>& element_id, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    if (not get<LinearSolver::Schwarz::Tags::SkipSubdomainSolverResets<
+            OptionsGroup>>(box)) {
+      if (UNLIKELY(get<logging::Tags::Verbosity<OptionsGroup>>(box) >=
+                   ::Verbosity::Debug)) {
+        Parallel::printf("%s " + Options::name<OptionsGroup>() +
+                             ": Reset subdomain solver\n",
+                         element_id);
```

Comment:
`"%s %s:  Reset subdomain solver\n", element_id, Options::name<OptionsGroup>()` (So it doesn't segfault if `Options::name<OptionsGroup>()` happens to have a % in it.)